lib/pulfalight/traject/ead2_component_config.rb

```diff
@@ -288,6 +269,99 @@
   accumulator.concat context.output_hash["level_ssm"]&.map(&:capitalize)
 end
 
+to_field "container_location_codes_ssim" do |record, accumulator, context|
+  if context.output_hash["level_ssm"] == ["Text"]
+    # Text records have no container information, but can be requested. Copy the
+    # container info from the parent.
+    parent = context.clipboard[:parent] || settings[:parent]
+    accumulator.replace(parent.output_hash["container_location_codes_ssim"])
+  else
+    record.xpath("./did/container").each do |container_element|
+      container_location_code = container_element.attributes["altrender"].to_s
+      accumulator << container_location_code if container_location_code.present?
+    end
+  end
+end
+
+to_field "container_information_ssm" do |record, accumulator, context|
+  if context.output_hash["level_ssm"] == ["Text"]
+    # Text records have no container information, but can be requested. Copy the
+    # container info from the parent.
+    parent = context.clipboard[:parent] || settings[:parent]
+    accumulator.replace(parent.output_hash["container_information_ssm"])
+  else
+    record.xpath("./did/container").each do |container_element|
+      container_location_code = container_element.attributes["altrender"].to_s
+      container_profile = container_element.attributes["encodinganalog"].to_s
+      next if container_location_code.blank?
+      accumulator << {
+        location_code: container_location_code,
+        profile: container_profile
+      }.to_json
+    end
```

Comment:
why is this to_json?

Reply:
Because the two pieces of information need to be serialized together

Reply:
thank you